### PR TITLE
add 'lighter' and 'bolder' font weight classes

### PR DESCRIPTION
**Pull request by browner12**
these are 2 special values for font weight, that will give their content a font-weight value of 100 more or less than their inherited font-weight.

probably doesn't fully fulfill this issue, https://github.com/twbs/bootstrap/issues/23969, but it's a start

### CODE REVIEW

**Comment by browner12** [comment]
to fix these errors I guess we'd have to update

https://github.com/twbs/bootstrap/blob/v4-dev/.stylelintrc#L21

I can't speak to the original intent behind using this rule, so I'm not sure what the appropriate answer is.

**Comment by XhmikosR** [comment]
Don't bolder and lighter have numeric equivalents?

**Comment by patrickhlauke** [comment]
> Don't bolder and lighter have numeric equivalents?

they are relative to the inherited/current value, so no

**Comment by browner12** [comment]
'bold' is equivalent to 700.

'lighter' and 'bolder' are variable, and are either 100 more or less than the value they inherit.

```html
<div style="font-weight:600; ">
     <div style="font-weight:lighter; ">This text will be 500 weight.</div>
     <div style="font-weight:bolder; ">This text will be 700 weight.</div>
</div>
```

http://htmldog.com/references/css/properties/font-weight/

**Comment by XhmikosR** [comment]
Ok, it's passing, so it's just a matter if we want this in or not. /CC @mdo @andresgalante

**Comment by MartijnCuppens** [comment]
These utility classes might need more documentation. There will be more people who think these values have numeric equivalents.

**Comment by browner12** [comment]
sounds good. if it get's accepted i'll add some docs.

**Comment by XhmikosR** [comment]
It's already been accepted for v4.2 :)

**Comment by browner12** [comment]
really? link?

File: scss/utilities/_text.scss [hunk 1]
```diff
@@ -32,10 +32,12 @@
 
 // Weight and italics
 
-.font-weight-light  { font-weight: $font-weight-light !important; }
-.font-weight-normal { font-weight: $font-weight-normal !important; }
-.font-weight-bold   { font-weight: $font-weight-bold !important; }
-.font-italic        { font-style: italic !important; }
+.font-weight-light   { font-weight: $font-weight-light !important; }
+.font-weight-lighter { font-weight: lighter !important; }
```
Comment (ahtee) [inline scss/utilities/_text.scss]: would this need to stay uniform with the other styles and be 
lighter -> $font-weight-lighter,
bolder -> $font-weight-bolder, etc?

would there need to be additions to make this work?

Comment (XhmikosR) [reply scss/utilities/_text.scss]: Those are CSS values.

**Review by mdo — CHANGES_REQUESTED** [review]
Per the conversation at https://github.com/twbs/bootstrap/pull/26580/files#r193721911, let's make these variables so that folks can change them if they so choose. I know these values are inherently relative, but they could easily be remapped to `100` and `900` depending on the situation.

**Comment by mdo** [comment]
I understand their intended usage, but people do whatever they want with CSS like this. 😆

My point is, if we include two new variables here, we'll be helping folks who want to have explicit options here. As soon as these become utilities and variables, it's an opportunity to move away from browser default values.

Take this example below: both are valid in their own contexts.

```scss
.font-weight-lighter { font-weight-lighter: lighter; }
.font-weight-light { font-weight-lighter: light; }

.font-weight-lighter { font-weight-lighter: 100; }
.font-weight-light { font-weight-lighter: 300; }
```

**Comment by browner12** [comment]
alright, the variables have been added.

**Review by mdo — APPROVED** [review]
Thank you!

**Pull request merged**